src/Data/Array/Polarized.hs

```diff
@@ -0,0 +1,39 @@
+{-# OPTIONS_GHC -fno-warn-partial-type-signatures #-}
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+
+module Data.Array.Polarized
+  ( transfer
+  , walk
+  )
+  where
+
+-- XXX: there is lots of scope for testing array functions
+
+import qualified Data.Array.Destination as DArray
+import qualified Data.Array.Polarized.Pull.Internal as Pull
+import qualified Data.Array.Polarized.Pull as Pull
+import qualified Data.Array.Polarized.Push as Push
+import Prelude.Linear
+
+import Data.Vector (Vector)
+
+
+-- See:
+--
+-- - http://lopezjuan.com/limestone/vectorcomp.pdf
+-- - http://jyp.github.io/posts/controlled-fusion.html
+--
+-- The general spirit is: `Array` are those arrays which are friendly in
+-- returned-value possition. And `PullArray` are those arrays which are friendly
```

Comment:
```suggestion
-- returned-value position. And `PullArray` are those arrays which are friendly
```

The whole documentation is fairly out of date, though. But it contains an nice intuition :slightly_smiling_face: .